core/include/gnuradio-4.0/Port.hpp

```diff
@@ -568,6 +565,24 @@ struct Port {
 
     [[nodiscard]] constexpr static bool isOptional() noexcept { return kIsOptional; }
 
+    [[nodiscard]] constexpr std::size_t nReaders() const noexcept {
+        if constexpr (kIsInput) {
+            return -1UZ;
+        } else {
+            return _ioHandler.buffer().n_readers() > 0;
+        }
+    }
+
+    [[nodiscard]] constexpr std::size_t nWriters() const noexcept {
+        if constexpr (kIsInput) {
+            return _ioHandler.buffer().n_writers();
+        } else {
+            return -1UZ;
```

Comment:
Why, `-1UZ`? Why not `0`, or, if this is considered an error, add a failing static_assert?

Reply:
See reply to Alex above